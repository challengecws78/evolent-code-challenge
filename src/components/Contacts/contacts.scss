.tableContainer {
    padding-top: 1em;
    .contactTable {
        width: 60%;
        margin: 0 auto;
        padding: 10px;
    }
    .no-results {
        text-align: center;
    }
}
  
.contact-input {
    margin-top: 50px;
    button {
        display: block;
        margin: 0 auto;
    }
}

table {
    border: 1px solid #333;
    border-collapse: collapse;

    th {
        padding: 5px;
        margin: 0;
        text-align: center;
        font-weight: normal;
        font-size: 15px;
    }

    td {
        text-align: center;
        padding: 5px;
    }

    thead {
        background-color: #bbb;
        th {
            border-bottom: 1px solid #ccc;
        }
    } 
}

.contact-form {
    display: block;
    width: 300px;
    margin: 0 auto;
}

h1, h2 {
    text-align: center;
}

a {
    display: inline-block;
    padding: 20px;
    margin: 0 auto;
}

header {
    width: 100%;
    color: #fff;
    background: #333;
    nav {
        color: #fff;
        text-align: center;
    }
}

body {
    margin: 0px;
}

table tbody tr:hover {
    background-color: #fff;
}
  
table tbody tr:hover td:hover {
    background-color: #333;
    color: #fff;
}

footer {
    height: 40px;
    width: 100%;
    text-align: center;
    font-family: 'Rokkitt', serif;
    text-shadow: 2px 2px #FFF;
    font-size: 15px;
    margin-top: 10em;
    border-top: 1px solid #000;
    padding: 10px 0px 10px 0px;
}

.filters {
    margin-top: 2em; 
    fieldset{
        border: 1px double #000;
        width: 50%;
        margin: 0 auto;
        padding: 20px 0 20px 0;
        input, label {
            text-align: center;
            display: block;
            margin: 0 auto;
        }
        input {
            width: 50%;
        }
        button {
            display: block;
            margin: 0 auto;
        }
        .sort {
            display:flex;
            align-items: center;
            justify-content: center;
        }
    }
}

