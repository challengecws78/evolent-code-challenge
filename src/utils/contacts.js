export const contacts = [
    {
      "id": "1",
      "firstName": "Vergie",
      "lastName": "Vankirk",
      "email": "Vergie.Vankirk@gmailio.com",
      "phone": "(374) 684-6831",
      "status": "active"
    },
    {
      "id": "2",
      "firstName": "Milford",
      "lastName": "Mizzell",
      "email": "milfordmizzell@gmailio.com",
      "phone": "(505) 2812342",
      "status": "active"
    },
    {
      "id": "3",
      "firstName": "Magen",
      "lastName": "Mart",
      "email": "magen.mart@eggmail.com",
      "phone": "(757) 702-8058",
      "status": "active"
    },
    {
      "id": "4",
      "firstName": "Kieth",
      "lastName": "Kalper",
      "email": "keith.kalpz@gmailio.com",
      "phone": "(168) 374-6980",
      "status": "active"
    },
    {
      "id": "5",
      "firstName": "Felton",
      "lastName": "Feit",
      "email": "feltonfeit@eggmail.com",
      "phone": "(483) 695-7967",
      "status": "active"
    },
];