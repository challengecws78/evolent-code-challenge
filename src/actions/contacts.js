import * as types from '../utils/types';
import uuid from 'uuid';

export const fetchContacts = contacts => ({
    type: types.FETCH_CONTACTS,
    contacts
});

export const addContact = contact => ({
    type: types.ADD_CONTACT,
    contact: {
        id: uuid(),
        ...contact
    }
});

export const updateContact = (id , contact) => ({
    type: types.UPDATE_CONTACT,
    id,
    contact
});

export const removeContact = (id, contact) => ({
    type: types.REMOVE_CONTACT,
    id,
    contact
});