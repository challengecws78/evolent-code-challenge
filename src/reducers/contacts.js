import * as types from '../utils/types';

export default (state=[], action) => {
    switch(action.type) {
        case types.FETCH_CONTACTS:
            return action.contacts
        case types.ADD_CONTACT:
            return [
                ...state,
                action.contact
            ]
        case types.UPDATE_CONTACT:
            return state.map(contact => {
                if(contact.id === action.id){
                    return {
                      ...contact,
                      ...action.contact
                    }
                } else {
                    return contact
                }
            });
        case types.REMOVE_CONTACT:
            return state.map(contact => {
                if(contact.id === action.id){
                    return {
                    ...contact,
                    ...action.contact
                    }
                } else {
                    return contact
                }
            });
        default:
            return state;
    }
}