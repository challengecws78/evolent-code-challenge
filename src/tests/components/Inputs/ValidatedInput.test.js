import React from 'react'
import { ValidatedInput } from '../../../components/Inputs/ValidatedInput';
import { shallow } from 'enzyme';
import { contacts } from '../../fixtures/contacts';

test('should render ValidatedInput with an empty value', ()=> {
    const onChange = jest.fn();
    const validate = jest.fn();
    const wrapper = shallow(
        <ValidatedInput 
            label="First Name" 
            name="firstName"
            type="text"
            placeholder="First Name"
            value=''
            validate={validate}
            onChange={onChange}
        />
    ); 
    expect(wrapper).toMatchSnapshot();
});

test('should render with value', ()=> {
    const onChange = jest.fn();
    const validate = jest.fn();
    const wrapper = shallow(
        <ValidatedInput
            label="First Name" 
            name="firstName"
            type="text"
            placeholder="First Name"
            value={contacts[0].name}
            validate={validate}
            onChange={onChange}
        />
    ); 
    expect(wrapper).toMatchSnapshot();
});

test('should throw error if name is empty', ()=> {
    const onChange = jest.fn();
    const name = 'firstName'
    const value = '';
    const wrapper = shallow(
        <ValidatedInput
            label="First Name" 
            name={name}
            type="text"
            placeholder="First Name"
            value={value}
            validate={val => (val ? false : 'First Name is Required')}
            onChange={onChange}
        />
    ); 
    wrapper.find('input').simulate('change', {
      target: { name, value }
    });
    expect(wrapper.state('error')).toBe('First Name is Required');
});

test('should not error if name is not empty', ()=> {
    const onChange = jest.fn();
    const name = 'firstName'
    const value = 'Aaron';
    const wrapper = shallow(
        <ValidatedInput
            label="First Name" 
            name={name}
            type="text"
            placeholder="First Name"
            value={value}
            validate={val => (val ? false : 'First Name is Required')}
            onChange={onChange}
        />
    ); 
    wrapper.find('input').simulate('change', {
      target: { name, value }
    });
    expect(wrapper.state('error')).toBe(false);
});