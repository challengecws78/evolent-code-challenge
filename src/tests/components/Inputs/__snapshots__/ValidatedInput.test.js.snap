// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ValidatedInput with an empty value 1`] = `
<div
  className="form-group"
>
  <label>
    First Name
  </label>
  <input
    className="form-control"
    name="firstName"
    onChange={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />
  <span
    style={
      Object {
        "color": "red",
      }
    }
  />
</div>
`;

exports[`should render with value 1`] = `
<div
  className="form-group"
>
  <label>
    First Name
  </label>
  <input
    className="form-control"
    name="firstName"
    onChange={[Function]}
    placeholder="First Name"
    type="text"
  />
  <span
    style={
      Object {
        "color": "red",
      }
    }
  />
</div>
`;
